processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/column/ArrayDoubleGroupByColumnSelectorStrategy.java

```diff
@@ -53,15 +41,13 @@ protected int computeDictionaryId(ColumnValueSelector selector)
     if (object == null) {
       return GROUP_BY_MISSING_VALUE;
     } else if (object instanceof Double) {
-      return addToIndexedDictionary(ImmutableList.of((Double) object));
+      return addToIndexedDictionary(new Object[]{object});
     } else if (object instanceof List) {
-      return addToIndexedDictionary((List<Double>) object);
+      return addToIndexedDictionary(((List) object).toArray());
     } else if (object instanceof Double[]) {
-      return addToIndexedDictionary(Arrays.asList((Double[]) object));
+      return addToIndexedDictionary(Arrays.stream((Double[]) object).toArray());
```

Comment:
i wonder if we need to handle this case anymore (we used to not homogenize arrays to `Object[]` prior to #12914). Same comment for other types

That said, it probably doesn't hurt much to be here... just might also be able to get by with removing it

Reply:
Since there are ad-hoc checks elsewhere, I kept it for now just in case we do encounter a rogue selector (possibly in some custom extension). If we do shift to ArrayColumnValueSelectors, we can get rid of it safely. I have added a comment about the redundancy of the branch though.